tiledb/sm/query/writer.cc

```diff
@@ -2548,10 +2561,11 @@ Status Writer::prepare_full_tiles_var(
   auto buffer = (uint64_t*)it->second.buffer_;
   auto buffer_var = (unsigned char*)it->second.buffer_var_;
   auto buffer_validity = (uint8_t*)it->second.validity_vector_.buffer();
-  auto buffer_size = it->second.buffer_size_;
+  auto buffer_size =
+      get_offset_buffer_size(it->first, *it->second.buffer_size_);
   auto buffer_var_size = it->second.buffer_var_size_;
   auto capacity = array_schema_->capacity();
-  auto cell_num = *buffer_size / constants::cell_var_offset_size;
+  auto cell_num = buffer_size / constants::cell_var_offset_size;
```

Comment:
By L2565, I assume the offsets path may contain either 32-bit or 64-bit offsets.

If this is true, `constants::cell_var_offset_size` on this line must be replaced by `offsets_bytesize()`. If this is not true, `get_offset_buffer_size()` should always subtract 64-bits instead of `offsets_bytesize()` in the calculation.

Reply:
Right, I fixed it to subtract 64 bits instead